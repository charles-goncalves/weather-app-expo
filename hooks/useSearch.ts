import { useState } from "react";
import useWeatherAPI from "./useWeatherAPI";
import {
  WEATHER_API_CURRENT_URL,
  WEATHER_API_SEARCH_URL,
} from "@/services/constants";
import { TemperatureUnit } from "@/components/ui/TemperatureUnitSelector";

interface IForecastAPIResponse {
  location: {
    name: string;
    region: string;
    country: string;
  };
  current: {
    temp_c: number;
    temp_f: number;
    condition: {
      text: string;
      icon: string;
    };
  };
  forecast: {
    forecastday: {
      day: {
        maxtemp_c: number;
        maxtemp_f: number;
        mintemp_c: number;
        mintemp_f: number;
      };
      hour: {
        temp_f: number;
        time: string;
        time_epoch: number;
        condition: {
          text: string;
          icon: string;
        };
        temp_c: number;
      }[];
    }[];
  };
}

interface ISearchAPIResponseItem {
  id: number;
  name: string;
}

const useSearch = () => {
  const [query, setQuery] = useState<string>("");
  const [unit, setUnit] = useState<TemperatureUnit>("C");
  const [selectedLocation, setSelectedLocation] = useState<string | null>(null);
  const [suspendedListEnabled, setSuspendedListEnabled] =
    useState<boolean>(true);

  const {
    weatherData: suggestions,
    loading: suggestionsLoading,
    error: suggestionsError,
    submit: suggestionSubmit,
  } = useWeatherAPI<ISearchAPIResponseItem[]>(
    WEATHER_API_SEARCH_URL,
    query ? { q: query } : undefined
  );

  const {
    weatherData: currentWeather,
    loading: currentLoading,
    error: currentError,
    submit: currentSubmit,
  } = useWeatherAPI<IForecastAPIResponse>(
    WEATHER_API_CURRENT_URL,
    selectedLocation ? { q: selectedLocation, days: "2" } : undefined
  );

  const handleSearch = (text: string) => {
    setQuery(text);
    setSuspendedListEnabled(true);
    if (text.length > 2) {
      suggestionSubmit();
    }
  };

  const handleSuggestionPress = (location: string) => {
    setQuery(location);
    setSelectedLocation(location);
    setSuspendedListEnabled(false);
    currentSubmit();
  };

  const handleForecastList = (currentHour: number) => {
    if (!currentWeather) return [];

    const currentDayHours = currentWeather.forecast.forecastday[0].hour.filter(
      (item) => {
        const forecastDate = new Date(item.time);
        const forecastHour = forecastDate.getHours();

        if (forecastHour <= currentHour || forecastHour > currentHour + 5)
          return undefined;

        return item;
      }
    );

    const amountNextDay = 5 - currentDayHours.length;
    const nextDayHours = amountNextDay
      ? currentWeather.forecast.forecastday[1].hour.filter((item) => {
          const forecastDate = new Date(item.time);
          const forecastHour = forecastDate.getHours();

          if (forecastHour >= amountNextDay) return undefined;

          return item;
        })
      : [];

    return currentDayHours.concat(nextDayHours);
  };

  return {
    query,
    unit,
    suggestions,
    suggestionsLoading,
    suggestionsError,
    currentWeather,
    currentLoading,
    currentError,
    suspendedListEnabled,
    handleForecastList,
    handleSearch,
    handleSuggestionPress,
    handleUnitChange: (unit: TemperatureUnit) => setUnit(unit),
  };
};

export default useSearch;
