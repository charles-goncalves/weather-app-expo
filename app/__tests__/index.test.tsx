import React from "react";
import { render, fireEvent, waitFor } from "@testing-library/react-native";
import WeatherScreen from "../index";
import useSearch from "../../hooks/useSearch";

jest.mock("../../hooks/useSearch");

describe("WeatherScreen", () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it("renders correctly", () => {
    (useSearch as jest.Mock).mockReturnValue({
      query: "",
      unit: "C",
      suggestions: [],
      suggestionsLoading: false,
      suggestionsError: null,
      currentWeather: null,
      currentLoading: false,
      currentError: null,
      suspendedListEnabled: true,
      handleForecastList: jest.fn(),
      handleSearch: jest.fn(),
      handleSuggestionPress: jest.fn(),
      handleUnitChange: jest.fn(),
    });

    const { toJSON } = render(<WeatherScreen />);
    expect(toJSON()).toMatchSnapshot();
  });

  it("updates input value and triggers search", async () => {
    const mockHandleSearch = jest.fn();
    (useSearch as jest.Mock).mockReturnValue({
      query: "",
      unit: "C",
      suggestions: [],
      suggestionsLoading: false,
      suggestionsError: null,
      currentWeather: null,
      currentLoading: false,
      currentError: null,
      suspendedListEnabled: true,
      handleForecastList: jest.fn(),
      handleSearch: mockHandleSearch,
      handleSuggestionPress: jest.fn(),
      handleUnitChange: jest.fn(),
    });

    const { getByPlaceholderText } = render(<WeatherScreen />);
    const input = getByPlaceholderText("Enter a location...");

    fireEvent.changeText(input, "New York");
    fireEvent(input, "submitEditing");

    await waitFor(() => {
      expect(mockHandleSearch).toHaveBeenCalledWith("New York");
    });
  });

  it("renders search suggestions and handles selection", async () => {
    const mockHandleSuggestionPress = jest.fn();
    (useSearch as jest.Mock).mockReturnValue({
      query: "",
      unit: "C",
      suggestions: [{ id: 1, name: "New York" }],
      suggestionsLoading: false,
      suggestionsError: null,
      currentWeather: null,
      currentLoading: false,
      currentError: null,
      suspendedListEnabled: true,
      handleForecastList: jest.fn(),
      handleSearch: jest.fn(),
      handleSuggestionPress: mockHandleSuggestionPress,
      handleUnitChange: jest.fn(),
    });

    const { getByText } = render(<WeatherScreen />);

    const suggestion = getByText("New York");
    fireEvent.press(suggestion);

    expect(mockHandleSuggestionPress).toHaveBeenCalledWith("New York");
  });

  it("renders the weather data correctly", () => {
    (useSearch as jest.Mock).mockReturnValue({
      query: "",
      unit: "F",
      suggestions: [],
      suggestionsLoading: false,
      suggestionsError: null,
      currentWeather: {
        location: { name: "New York", country: "USA", region: "NY" },
        current: {
          temp_c: 25,
          temp_f: 77,
          condition: { text: "Sunny", icon: "//cdn.weather.com/icon.png" },
        },
        forecast: {
          forecastday: [
            {
              day: {
                maxtemp_c: 30,
                maxtemp_f: 86,
                mintemp_c: 20,
                mintemp_f: 68,
              },
            },
          ],
        },
      },
      currentLoading: false,
      currentError: null,
      suspendedListEnabled: false,
      handleSearch: jest.fn(),
      handleForecastList: jest.fn().mockReturnValue([
        {
          time: "2025-02-09 22:00",
          time_epoch: 1,
          temp_c: 20,
          temp_f: 68,
          condition: { text: "Clear", icon: "//icon.png" },
        },
        {
          time: "2025-02-09 23:00",
          time_epoch: 2,
          temp_c: 19,
          temp_f: 66,
          condition: { text: "Cloudy", icon: "//icon.png" },
        },
        {
          time: "2025-02-09 00:00",
          time_epoch: 3,
          temp_c: 18,
          temp_f: 64,
          condition: { text: "Rainy", icon: "//icon.png" },
        },
      ]),
      handleSuggestionPress: jest.fn(),
      handleUnitChange: jest.fn(),
    });

    const { toJSON } = render(<WeatherScreen />);

    expect(toJSON()).toMatchSnapshot();
  });

  it("renders and interacts with unit selector", () => {
    const mockHandleUnitChange = jest.fn();
    (useSearch as jest.Mock).mockReturnValue({
      query: "",
      unit: "C",
      suggestions: [],
      suggestionsLoading: false,
      suggestionsError: null,
      currentWeather: null,
      currentLoading: false,
      currentError: null,
      suspendedListEnabled: true,
      handleSearch: jest.fn(),
      handleForecastList: jest.fn(),
      handleSuggestionPress: jest.fn(),
      handleUnitChange: mockHandleUnitChange,
    });

    const { getByText } = render(<WeatherScreen />);
    const fahrenheitButton = getByText("Fahrenheit (Â°F)");
    fireEvent.press(fahrenheitButton);

    expect(mockHandleUnitChange).toHaveBeenCalledWith("F");
  });

  it("renders and interacts with unit selector", () => {
    const mockHandleSearch = jest.fn();
    (useSearch as jest.Mock).mockReturnValue({
      query: "query",
      unit: "C",
      suggestions: [],
      suggestionsLoading: false,
      suggestionsError: null,
      currentWeather: null,
      currentLoading: false,
      currentError: null,
      suspendedListEnabled: true,
      handleForecastList: jest.fn(),
      handleSearch: mockHandleSearch,
      handleSuggestionPress: jest.fn(),
      handleUnitChange: jest.fn(),
    });

    const { getByTestId } = render(<WeatherScreen />);
    const fahrenheitButton = getByTestId("searchButton");
    fireEvent.press(fahrenheitButton);

    expect(mockHandleSearch).toHaveBeenCalledWith("query");
  });

  it("shows loading indicator when fetching data", () => {
    (useSearch as jest.Mock).mockReturnValue({
      query: "",
      unit: "C",
      suggestions: [],
      suggestionsLoading: true,
      suggestionsError: "mockError",
      currentWeather: null,
      currentLoading: true,
      currentError: "mockError",
      suspendedListEnabled: true,
      handleForecastList: jest.fn(),
      handleSearch: jest.fn(),
      handleSuggestionPress: jest.fn(),
      handleUnitChange: jest.fn(),
    });

    const { getAllByTestId } = render(<WeatherScreen />);
    expect(getAllByTestId("loading-indicator")).toHaveLength(2);
  });
});
