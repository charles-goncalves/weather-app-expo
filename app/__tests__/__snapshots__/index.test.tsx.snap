// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeatherScreen renders correctly 1`] = `
<RCTSafeAreaView
  style={
    {
      "backgroundColor": "#fff",
      "flex": 1,
      "padding": 16,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "marginHorizontal": 20,
        "marginVertical": 16,
      }
    }
  >
    <Text
      style={
        {
          "flex": 0.28,
          "fontSize": 16,
          "fontWeight": "bold",
          "marginRight": 5,
          "textAlign": "right",
        }
      }
    >
      Temperature Unit:
    </Text>
    <View
      style={
        {
          "flex": 0.8,
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#000",
            "borderColor": "#000",
            "borderRadius": 8,
            "borderWidth": 1,
            "flex": 1,
            "marginHorizontal": 4,
            "opacity": 1,
            "padding": 12,
          }
        }
      >
        <Text
          style={
            [
              {
                "color": "#aaa",
                "fontSize": 16,
              },
              {
                "color": "#fff",
              },
            ]
          }
        >
          Celsius (°C)
        </Text>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#f5f5f5",
            "borderColor": "#ccc",
            "borderRadius": 8,
            "borderWidth": 1,
            "flex": 1,
            "marginHorizontal": 4,
            "opacity": 1,
            "padding": 12,
          }
        }
      >
        <Text
          style={
            [
              {
                "color": "#aaa",
                "fontSize": 16,
              },
              false,
            ]
          }
        >
          Fahrenheit (°F)
        </Text>
      </View>
    </View>
  </View>
  <View>
    <TextInput
      onChangeText={[MockFunction]}
      onSubmitEditing={[Function]}
      placeholder="Enter a location..."
      style={
        {
          "backgroundColor": "#fff",
          "borderColor": "black",
          "borderRadius": 8,
          "borderWidth": 1,
          "height": 40,
          "margin": 20,
          "paddingHorizontal": 8,
        }
      }
      value=""
    />
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
          "position": "absolute",
          "right": 30,
          "top": 25,
        }
      }
      testID="searchButton"
    >
      <ViewManagerAdapter_SymbolModule
        animated={false}
        colors={[]}
        name="magnifyingglass.circle"
        resizeMode="scaleAspectFit"
        style={
          [
            {
              "height": 24,
              "width": 24,
            },
            [
              {
                "height": 30,
                "width": 30,
              },
              undefined,
            ],
          ]
        }
        tintColor="black"
        type="monochrome"
        weight="regular"
      />
    </View>
    <View
      style={
        {
          "backgroundColor": "white",
          "position": "absolute",
          "top": 60,
          "width": "100%",
          "zIndex": 10,
        }
      }
    >
      <RCTScrollView
        data={[]}
        getItem={[Function]}
        getItemCount={[Function]}
        keyExtractor={[Function]}
        onContentSizeChange={[Function]}
        onLayout={[Function]}
        onMomentumScrollBegin={[Function]}
        onMomentumScrollEnd={[Function]}
        onScroll={[Function]}
        onScrollBeginDrag={[Function]}
        onScrollEndDrag={[Function]}
        removeClippedSubviews={false}
        renderItem={[Function]}
        scrollEventThrottle={0.0001}
        stickyHeaderIndices={[]}
        style={
          {
            "borderLeftWidth": 1,
            "borderRightWidth": 1,
            "marginHorizontal": 20,
            "maxHeight": 250,
          }
        }
        viewabilityConfigCallbackPairs={[]}
      >
        <View />
      </RCTScrollView>
      <Text
        style={
          {
            "color": "#666",
            "fontSize": 16,
            "marginTop": 16,
            "textAlign": "center",
          }
        }
      >
        No results found.
      </Text>
    </View>
  </View>
  <RCTScrollView>
    <View />
  </RCTScrollView>
</RCTSafeAreaView>
`;

exports[`WeatherScreen renders the weather data correctly 1`] = `
<RCTSafeAreaView
  style={
    {
      "backgroundColor": "#fff",
      "flex": 1,
      "padding": 16,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "marginHorizontal": 20,
        "marginVertical": 16,
      }
    }
  >
    <Text
      style={
        {
          "flex": 0.28,
          "fontSize": 16,
          "fontWeight": "bold",
          "marginRight": 5,
          "textAlign": "right",
        }
      }
    >
      Temperature Unit:
    </Text>
    <View
      style={
        {
          "flex": 0.8,
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#000",
            "borderColor": "#000",
            "borderRadius": 8,
            "borderWidth": 1,
            "flex": 1,
            "marginHorizontal": 4,
            "opacity": 1,
            "padding": 12,
          }
        }
      >
        <Text
          style={
            [
              {
                "color": "#aaa",
                "fontSize": 16,
              },
              {
                "color": "#fff",
              },
            ]
          }
        >
          Celsius (°C)
        </Text>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#f5f5f5",
            "borderColor": "#ccc",
            "borderRadius": 8,
            "borderWidth": 1,
            "flex": 1,
            "marginHorizontal": 4,
            "opacity": 1,
            "padding": 12,
          }
        }
      >
        <Text
          style={
            [
              {
                "color": "#aaa",
                "fontSize": 16,
              },
              false,
            ]
          }
        >
          Fahrenheit (°F)
        </Text>
      </View>
    </View>
  </View>
  <View>
    <TextInput
      onChangeText={[MockFunction]}
      onSubmitEditing={[Function]}
      placeholder="Enter a location..."
      style={
        {
          "backgroundColor": "#fff",
          "borderColor": "black",
          "borderRadius": 8,
          "borderWidth": 1,
          "height": 40,
          "margin": 20,
          "paddingHorizontal": 8,
        }
      }
      value=""
    />
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
          "position": "absolute",
          "right": 30,
          "top": 25,
        }
      }
      testID="searchButton"
    >
      <ViewManagerAdapter_SymbolModule
        animated={false}
        colors={[]}
        name="magnifyingglass.circle"
        resizeMode="scaleAspectFit"
        style={
          [
            {
              "height": 24,
              "width": 24,
            },
            [
              {
                "height": 30,
                "width": 30,
              },
              undefined,
            ],
          ]
        }
        tintColor="black"
        type="monochrome"
        weight="regular"
      />
    </View>
  </View>
  <RCTScrollView>
    <View>
      <View
        style={
          {
            "margin": 20,
          }
        }
      >
        <Text
          style={
            {
              "fontSize": 24,
              "fontWeight": "bold",
              "marginBottom": 8,
            }
          }
        >
          Current Weather:
        </Text>
        <View
          style={
            {
              "alignItems": "center",
              "borderColor": "black",
              "borderRadius": 10,
              "borderWidth": 1,
              "flexDirection": "row",
              "justifyContent": "space-between",
              "marginVertical": 16,
              "padding": 12,
            }
          }
        >
          <View
            style={
              {
                "maxWidth": "70%",
              }
            }
          >
            <Text
              style={
                {
                  "fontSize": 20,
                  "fontWeight": "bold",
                }
              }
            >
              New York
              ,
               
              USA
            </Text>
            <Text
              style={
                {
                  "fontSize": 16,
                  "marginBottom": 8,
                }
              }
            >
              NY
            </Text>
            <Text
              style={
                {
                  "fontSize": 18,
                  "marginBottom": 8,
                }
              }
            >
              Sunny
            </Text>
            <Text
              style={
                {
                  "fontSize": 16,
                  "marginBottom": 8,
                }
              }
            >
              H:
               
              86°
              , 
              L:
               
              68°
            </Text>
          </View>
          <View
            style={
              {
                "alignItems": "center",
                "justifyContent": "flex-start",
              }
            }
          >
            <Image
              source={
                {
                  "uri": "https://cdn.weather.com/icon.png",
                }
              }
              style={
                {
                  "height": 90,
                  "width": 90,
                }
              }
            />
            <Text
              style={
                {
                  "fontSize": 20,
                }
              }
            >
              77°
            </Text>
          </View>
        </View>
        <Text
          style={
            {
              "fontSize": 24,
              "fontWeight": "bold",
              "marginBottom": 8,
            }
          }
        >
          Forecast:
        </Text>
        <View
          style={
            {
              "borderBottomWidth": 0,
              "borderColor": "black",
              "borderRadius": 10,
              "borderWidth": 1,
              "marginTop": 8,
            }
          }
        >
          <View
            style={
              {
                "alignItems": "center",
                "borderBottomColor": "black",
                "borderBottomWidth": 1,
                "borderRadius": 10,
                "flexDirection": "row",
                "justifyContent": "space-between",
                "padding": 12,
              }
            }
          >
            <View>
              <Text
                style={
                  {
                    "fontSize": 16,
                  }
                }
              >
                22:00
              </Text>
              <Text
                style={
                  {
                    "fontSize": 16,
                  }
                }
              >
                Clear
              </Text>
            </View>
            <View
              style={
                {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                }
              }
            >
              <Text
                style={
                  {
                    "fontSize": 16,
                  }
                }
              >
                68°
              </Text>
              <Image
                source={
                  {
                    "uri": "https://icon.png",
                  }
                }
                style={
                  {
                    "height": 30,
                    "marginLeft": 5,
                    "width": 30,
                  }
                }
              />
            </View>
          </View>
          <View
            style={
              {
                "alignItems": "center",
                "borderBottomColor": "black",
                "borderBottomWidth": 1,
                "borderRadius": 10,
                "flexDirection": "row",
                "justifyContent": "space-between",
                "padding": 12,
              }
            }
          >
            <View>
              <Text
                style={
                  {
                    "fontSize": 16,
                  }
                }
              >
                23:00
              </Text>
              <Text
                style={
                  {
                    "fontSize": 16,
                  }
                }
              >
                Cloudy
              </Text>
            </View>
            <View
              style={
                {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                }
              }
            >
              <Text
                style={
                  {
                    "fontSize": 16,
                  }
                }
              >
                66°
              </Text>
              <Image
                source={
                  {
                    "uri": "https://icon.png",
                  }
                }
                style={
                  {
                    "height": 30,
                    "marginLeft": 5,
                    "width": 30,
                  }
                }
              />
            </View>
          </View>
          <View
            style={
              {
                "alignItems": "center",
                "borderBottomColor": "black",
                "borderBottomWidth": 1,
                "borderRadius": 10,
                "flexDirection": "row",
                "justifyContent": "space-between",
                "padding": 12,
              }
            }
          >
            <View>
              <Text
                style={
                  {
                    "fontSize": 16,
                  }
                }
              >
                00:00
              </Text>
              <Text
                style={
                  {
                    "fontSize": 16,
                  }
                }
              >
                Rainy
              </Text>
            </View>
            <View
              style={
                {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                }
              }
            >
              <Text
                style={
                  {
                    "fontSize": 16,
                  }
                }
              >
                64°
              </Text>
              <Image
                source={
                  {
                    "uri": "https://icon.png",
                  }
                }
                style={
                  {
                    "height": 30,
                    "marginLeft": 5,
                    "width": 30,
                  }
                }
              />
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</RCTSafeAreaView>
`;
